localstack-core/localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -9,6 +9,7 @@
 
 from localstack import config
 from localstack.aws.connect import connect_to
+from localstack.config import CFN_STRING_REPLACEMENT_DENY_LIST
```

Comment:
If you import it directly, the monkeypatch in the test cases will not work anymore, as it is now a module attribute of the template deployer module.

To support the monkeypatching, you need to use it with `config.CFN_STRING_REPLACEMENT_DENY_LIST`, this way it will always reference the current `config` value. That is why the test fails, but presumably did not locally, as the `template_deployer` module was likely already imported in the CI run :)

Reply:
Yes, you are basically patching the module level variable in config - if you go the indirection via config.VAR, it will always resolve to the current value in that module, which was changed with monkey patch. You could also assign directly, but the monkey patch fixture takes care of restoring the value after the test, which is convenient.